src/gui/sdlmain.cpp

```diff
@@ -2899,27 +2877,57 @@ static SDL_Point window_bounds_from_label(const std::string& pref,
 	const int percent = [&] {
 		if (pref.starts_with('s')) {
 			return SmallPercent;
+
 		} else if (pref.starts_with('m') || pref == "default" ||
 		           pref.empty()) {
 			return MediumPercent;
+
 		} else if (pref.starts_with('l')) {
 			return LargePercent;
+
 		} else if (pref == "desktop") {
 			return 100;
+
 		} else {
-			LOG_WARNING("DISPLAY: Invalid 'windowresolution' setting: '%s', "
-			            "using 'default'",
-			            pref.c_str());
+			LOG_WARNING(
+			        "DISPLAY: Invalid 'windowresolution' setting: '%s', "
+			        "using 'default'",
+			        pref.c_str());
+
+			set_section_property_value("sdl", "windowresolution", "default");
```

Comment:
Contrary to my previous point about window position, this one looks good to me.  We're only updating the setting if it is invalid.  We're not updating it in `maybe_limit_requested_resolution` which checks runtime environment.